dspace-spring-rest/src/test/java/org/dspace/app/rest/builder/CollectionBuilder.java

```diff
@@ -78,6 +79,20 @@ public Collection build() {
         return collection;
     }
 
+    public static Item createTemplateItem(final Context context, final Collection collection) {
```

Comment:
Can you move this method to the `ItemBuilder` class? That should allow us to create collection template items with metadata:

```
Item templateItem = ItemBuilder.createTemplateItemForCollection(context, col1)
                                                    .withSubject("Mathematics")
                                                    .build();
```

In the `createTemplateItemForCollection` you need to set a flag on the `ItemBuilder` object (e.g. `archiveItem = false;`) to remember that we should not call `installItem` on executing `build()`.

Reply:
Hi @tomdesair. Thanks for the review. @abollini I had this method in ItemBuilder class originally but @abollini  requested me to move to it CollectionBuilder. ¿How shall we proceed?

Reply:
the templateItem is really a piece of the collection (like a logo) so conceptually I think is more appropriate to have it created in the collectio builder.
Moreover, due to how the builders are call during the cleanup phase we could end with the collectionbuilder removing the templateitem and the itembuilder failing as the item doesn't longer exist.
Having the template item managed/known by a single builder is imho easier.

@tomdesair To add metadata to the template item can we simply get it after the collection creation (.build())?

Reply:
I agree with @abollini here.  This method conceptually should remain in the `CollectionBuilder`, as Item Templates are *not* normal Items. They belong to Collections, and can only be created from Collections.

If we find a reason to move this method in the future (to support future features/code), then we can move it at that time. For now though, I think we should just leave it where it is.

Reply:
@ppmdo : could you add Javadocs to this `public` method?  I notice it's missing Javadocs. Per our code style, we recommend Javadocs on all public methods/classes: https://wiki.duraspace.org/display/DSPACE/Code+Style+Guide